scripts/irods/test/test_misc.py

```diff
@@ -317,3 +319,29 @@ def test_server_correctly_cleans_up_proc_files__issue_6231(self):
         # Show that the proc directory will eventually become empty.
         lib.delayAssert(lambda: len(os.listdir(paths.proc_directory())) == 0)
 
+    def test_iinit_does_not_crash_from_long_home_directory__issue_5411(self):
+        with tempfile.TemporaryDirectory() as temp_dir_name:
+            # Create new home directory for test
+            temp_home = os.path.join(temp_dir_name, 'crazy', 'long', 'temp', 'home', 'directory',
+                                     'like', 'this', 'is', 'longer', 'than', 'the', 'solos', 'in', 'chameleon')
+            os.makedirs(temp_home)
+
+            # Setup environment
+            temp_env = os.environ.copy()
+            temp_env['HOME'] = temp_home
+
+            # Execute iinit
+            user_input = os.linesep.join(
+                ['localhost', '1247', 'rods', 'tempZone', 'rods']) + os.linesep
+            res = subprocess.run('iinit', input=user_input.encode(
+            ), stdout=subprocess.PIPE, stderr=subprocess.PIPE, env=temp_env)
```

Comment:
Let's make this one line. I'm guessing your editor's formatter caused the split at `.encode(`?

Why `subprocess.run()` instead of one of the `lib.execute*` functions?

Reply:
Yep. When I wrote that code originally, it was in one line. I formatted the code before committing, so what you see is what the formatter chose.

From my understanding, lib.execute* functions end up calling `execute_command_nonblocking(...)` which masks the `env` parameter that is needed for this test. So, I went with the `run()` function instead.

Reply:
Are there any downsides to using `.run()`?

Reply:
In this case, I don't believe so. In general, it's a bit less flexible than options such as `Popen`.

The [Python docs](https://docs.python.org/3.6/library/subprocess.html) recommend preferring the `.run()` function compared to other `subprocess` options:
> The recommended approach to invoking subprocesses is to use the `run()` function for all use cases it can handle. For more advanced use cases, the underlying `Popen` interface can be used directly.